x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/type_guards.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  TrustedAppCreatePending,
+  TrustedAppsListPageState,
+  TrustedAppCreateFailure,
+  TrustedAppCreateSuccess,
+} from './trusted_apps_list_page_state';
+import {
+  Immutable,
+  NewTrustedApp,
+  WindowsConditionEntry,
+} from '../../../../../common/endpoint/types';
+import { TRUSTED_APPS_SUPPORTED_OS_TYPES } from '../../../../../common/endpoint/constants';
+
+type CreateViewPossibleStates =
+  | TrustedAppsListPageState['createView']
+  | Immutable<TrustedAppsListPageState['createView']>;
+
+export const isTrustedAppCreatePendingState = (
```

Comment:
is this related to what bohdan was talking about before about having selectors return types?

Reply:
yes :)

Type guards are a runtime (executable code) function whose purpose is to ensure we are working with the correct type. In TS, it helps tell it that if this function evaluates to `true`, then the value being evaluated (`data`) is of a given type (note the `: data is TrustedAppCreateSuccess` in the return value for this function's signature.